### ISSUE TITLE
Exception when user tries to view to group management tab

### ISSUE THREAD

**Issue by rorydavidson** [sillycat/confluence-space-user-management#19]
Running Confluence 5.9.1 - when a user, who was not the owner/creator of the space, yet has full space admin permissions, tries to go the to Manage Users/Groups tab in space admin, they get the following error:
```
org.apache.velocity.exception.MethodInvocationException: Invocation of method 'getSpacesAsSpaceAdminForUser' in class csum.confluence.permissionmgmt.CustomPermissionManagerAction threw exception java.lang.NoSuchMethodError: com.atlassian.confluence.spaces.persistence.dao.SpaceDao.getSpaces(Lcom/atlassian/confluence/spaces/SpacesQuery;)Ljava/util/List; at /templates/permissionmgmt/display.vm[line 227, column 49]
    at org.apache.velocity.runtime.parser.node.ASTIdentifier.execute(ASTIdentifier.java:237)
caused by: java.lang.NoSuchMethodError: com.atlassian.confluence.spaces.persistence.dao.SpaceDao.getSpaces(Lcom/atlassian/confluence/spaces/SpacesQuery;)Ljava/util/List;
    at csum.confluence.permissionmgmt.CustomPermissionManagerAction.getSpacesAssociatedToUserForGivenPermission(CustomPermissionManagerAction.java:778)
```

**Comment by garysweaver** [comment]
@rorydavidson thanks for the heads up! Will take a look tonight and will remove the compatibility for that version for now.

**Comment by NicolasEsteves** [comment]
Hello!

Any news about this ticket?
Same issue for me, with Confluence 6.0.0.

{code}
org.apache.velocity.exception.MethodInvocationException: Invocation of method 'getSpacesAsSpaceAdminForUser' in class csum.confluence.permissionmgmt.CustomPermissionManagerAction threw exception java.lang.NoSuchMethodError: com.atlassian.confluence.spaces.persistence.dao.SpaceDao.getSpaces(Lcom/atlassian/confluence/spaces/SpacesQuery;)Ljava/util/List; at /templates/permissionmgmt/display.vm[line 227, column 49]
    at org.apache.velocity.runtime.parser.node.ASTIdentifier.execute(ASTIdentifier.java:237)
caused by: java.lang.NoSuchMethodError: com.atlassian.confluence.spaces.persistence.dao.SpaceDao.getSpaces(Lcom/atlassian/confluence/spaces/SpacesQuery;)Ljava/util/List;
    at csum.confluence.permissionmgmt.CustomPermissionManagerAction.getSpacesAssociatedToUserForGivenPermission(CustomPermissionManagerAction.java:778)
{code}

I understand that this is a compatibility issue between the plugin and Confluence. But I'd like to know if a new version is planned or not. Thank you in advance.

Regards,
Nicolas.

**Comment by garysweaver** [comment]
Sorry, CSUM is not compatible with 5.9.1 or 6.x currently. Hoping to add 5.9.x compatibility sooner. 6.x I'm not sure about yet. If you'd like to contribute, feel free.

**Comment by NicolasEsteves** [comment]
Hello @garysweaver,

Thank you for your quick answer. I know that almost none plugin is marked as compatible with Confluence 6.0.0 (OD), but it appears that all plugins that I've tested marked compatible with 5.9.2 work properly with Confluence 6.0.0. I assume the core is the same.

So, I'm going to wait for your next release in order to test it.

I'd like to contribute but I'm sorry to tell you that my development skills are not pretty good.

Regards,
Nicolas.

**Comment by NicolasEsteves** [comment]
Hello @garysweaver,

Just another question; do you have an ETA to give me? Are we talking about days, weeks... month? I ask you because it's a critical issue for my customer. Any details would be much appreciated.

Thank you in advance.

Regards,
Nicolas.

**Comment by sebValiantys** [comment]
Hi,
I am looking at upgrading the plugin to Confluence 5.9.2 right now and am stumbling on many dependency related issues. The last one jax-rpc, seems to be absent from public maven repos. Any hints on how to resolve it ? 

One remark, from the names of the files involved, it's used for SOAP services, which have been removed from the latest versions of Atlassian products in favor of REST apis. Is it kept for retro compatibility ? Since JIRA 6 the creation of users and groups is available in the REST api and JIRA 7 doesn't have a SOAP interface any more (https://developer.atlassian.com/blog/2015/05/jira-soap-api-announcement/) Should it be kept or should it be replaced ?

**Comment by sebValiantys** [comment]
Never mind my search for jax-rpc, finally managed to package the jar.

**Comment by garysweaver** [comment]
:thumbsup: PR #20. Will plan to build/release this weekend. Thanks much @sebValiantys !!

**Comment by garysweaver** [comment]
Released as v2.4.0.
